pkg/orchestration/wiring/edge/types.go

```diff
@@ -0,0 +1,46 @@
+package edge
+
+import "github.com/atlassian/voyager"
+
+type InstanceParameters struct {
+	ServiceName voyager.ServiceName `json:"serviceName"`
+	Resource    ResourceParameters  `json:"resource"`
+}
+
+type ResourceParameters struct {
+	Attributes Attributes `json:"attributes"`
+}
+
+type Attributes struct {
+	UpstreamAddress []UpstreamAddress `json:"upstream_address"`
+	UpstreamPort    int32             `json:"upstream_port,omitempty"`
```

Comment:
So the issue here is that the platform user will have parameters with different style for different resource types. As a user, I don't care if it's the platform or its the provider who specifies names. I see that naming is all over the place and I'm not amused by that. FWIW our provider contract page recommends `camelCase` to follow the existing style and Kubernetes style.

@roaanv @wathrog, do we want to do something about it?

Reply:
We have `snake_case`, we have `CapitalCase` and we have `camelCase`.

Reply:
Apart from aesthetics, I don't see this being a problem. The closer resource parameters are to the actual OSB contract, the easier it is to implement an autowiring function, and we probably want to make implementing autowiring functions as simple as possible...

Reply:
"Try to make it pass through" is actually our recommendation, yes. Yes, only aesthetics suffer but that is a significant issue imo.

Reply:
@cetanu  as discussed with @ash2k in person, the feedback from @roaanv is that we enforce the consistent style in SD, which means that we need to define separate types for SD, and convert to OSB contract internally. I will give it a shot.